doc/userguide/rules/http-keywords.rst

```diff
@@ -1,813 +1,1200 @@
-like ``distance``, ``offset``, ``nocase``, ``within``, etc.
+The ``http.request_body`` keyword is used to match on the HTTP request body
+that can be present in an HTTP request.
 
-Example of ``http.request_body`` in a HTTP request:
+It is possible to use any of the :doc:`payload-keywords` with the
+``http.request_body`` keyword.
 
-.. image:: http-keywords/client_body.png
+Example HTTP Request::
 
-Example of the purpose of ``http.client_body``:
+  POST /suricata.php HTTP/1.1
+  Content-Type: application/x-www-form-urlencoded
+  Host: suricata.io
+  Content-Length: 23
+  Connection: Keep-Alive
 
-.. image:: http-keywords/client_body1.png
+  Suricata request body
 
-Note: how much of the request/client body is inspected is controlled
-in the :ref:`libhtp configuration section
-<suricata-yaml-configure-libhtp>` via the ``request-body-limit``
-setting.
+.. container:: example-rule
 
-``http.request_body`` replaces the previous keyword name: ```http_client_body``. You may continue
-+to use the previous name, but it's recommended that rules be converted to use
-+the new name.
+  alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"HTTP Request Body Example"; \
+  flow:established,to_server; :example-rule-options:`http.request_body; \
+  content:"Suricata request body";` classtype:bad-unknown; sid:115; rev:1;)
+
+.. note:: How much of the request/client body is inspected is controlled
+  in the :ref:`libhtp configuration section
+  <suricata-yaml-configure-libhtp>` via the ``request-body-limit``
+  setting.
+
+.. note:: ``http.request_body`` replaces the previous keyword name,
+  ``http_client_body``. ``http_client_body`` can still be used but it is
+  recommended that rules be converted to use ``http.request_body``.
+
+.. _http.stat_code:
 
 http.stat_code
 --------------
 
-With the ``http.stat_code`` sticky buffer, it is possible to match
-specifically and only on the HTTP status code buffer. The keyword can
-be used in combination with all previously mentioned content modifiers
-like ``distance``, ``offset``, ``nocase``, ``within``, etc.
+The ``http.stat_code`` keyword is used to match on the HTTP status code
+that can be present in an HTTP response.
+
+It is possible to use any of the :doc:`payload-keywords` with the
+``http.stat_code`` keyword.
 
-Example of ``http.stat_code`` in a HTTP response:
+Example HTTP Response::
+
+  HTTP/1.1 200 OK
+  Content-Type: text/html
+  Server: nginx/0.8.54
+
+.. container:: example-rule
 
-.. image:: http-keywords/stat_code.png
+  alert http $EXTERNAL_NET any -> $HOME_NET any (msg:"HTTP Stat Code Response \
+  Example"; flow:established,to_client; :example-rule-options:`http.stat_code; \
+  content:"200";` classtype:bad-unknown; sid:117; rev:1;)
 
-Example of the purpose of ``http.stat_code``:
+.. note:: ``http.stat_code`` does not include the leading or trailing space
 
-.. image:: http-keywords/stat-code1.png
+.. _http.stat_msg:
 
 http.stat_msg
```

Comment:
Is there any chance we can get a note that reflects the behavior of http.stat_msg on HTTP/2 traffic. 

https://redmine.openinfosecfoundation.org/issues/4067#note-37
```
Philippe Antoine wrote in [#note-37](https://redmine.openinfosecfoundation.org/issues/4067#note-37):
>
>The empty buffer is http_stat_msg for HTTP2
```

Commit creating the constant empty buffer: 
https://github.com/OISF/suricata/commit/22fb385e23991f0a7e3b83aeb2693d8ab21bdf41